src/CMakeLists.txt

```diff
@@ -162,12 +166,14 @@ if (BUILD_SDF_TEST)
       endif()
       target_link_libraries(UNIT_parser_urdf_TEST PRIVATE ${URDF_LIBRARIES})
     endif()
+    target_link_libraries(UNIT_parser_urdf_TEST PRIVATE
+      ${TinyXML2_LIBRARIES})
   endif()
 endif()
 
 sdf_add_library(${sdf_target} ${sources})
 target_compile_features(${sdf_target} PUBLIC cxx_std_17)
-target_link_libraries(${sdf_target} PUBLIC
+target_link_libraries(${sdf_target} PRIVATE
   ${IGNITION-MATH_LIBRARIES}
```

Comment:
ign-math still needs to be public

Reply:
Fixed: 46ec586f27cd841a5530a24d6c3ec9922a18e44f